File: mc/rcmd/src/main/java/calebxzhou/rdi/mc/rcmd/RcmdDispatchResult.java
package calebxzhou.rdi.mc.rcmd;

public record RcmdDispatchResult(boolean found, RcmdResult result) {
    public static RcmdDispatchResult found(RcmdResult result) {
        return new RcmdDispatchResult(true, result == null ? RcmdResult.ok() : result);
    }

    public static RcmdDispatchResult notFound() {
        return new RcmdDispatchResult(false, RcmdResult.ok());
    }
}

File: mc/rcmd/src/main/kotlin/calebxzhou/rdi/mc/rcmd/RcmdDispatchResult.kt
package calebxzhou.rdi.mc.rcmd

data class RcmdDispatchResult(val found: Boolean, val result: RcmdResult) {
    fun found() = found

    fun result() = result

    companion object {
        @JvmStatic
        fun found(result: RcmdResult) = RcmdDispatchResult(true, result)

        @JvmStatic
        fun notFound() = RcmdDispatchResult(false, RcmdResult.ok())
    }
}
